samples/subsys/display/lvgl/src/main.c

```diff
@@ -15,6 +15,8 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(app);
 
+#undef GDEW027W3
```

Comment:
What's going on here? This is undefined, so the `#ifdef GDEW027W3` below is dead code.

If you need to match on a particular piece of hardware, then I think we need to get around to removing CONFIG_LVGL_DISPLAY_DEV_NAME and using a devicetree node instead.

Then you can either:

1. match on its compatible with DT_NODE_HAS_COMPAT: https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_NODE_HAS_COMPAT
2. add some other devicetree property that the driver or application can use to decide to use `display_refresh` instead of `display_blanking_off`

Reply:
@mbolivar-nordic  As suggested , Inorder match for this particular hardware, used DT_NODE_HAS_COMPAT.

Reply:
> If you need to match on a particular piece of hardware, then I think we need to get around to removing CONFIG_LVGL_DISPLAY_DEV_NAME

This part was important, can you please do that too?

What I mean is that Kconfig options which just specify a device name have been deprecate for a long time now that the devicetree.h API has been stabilized and the device model has been updated to use devicetree nodes.

The thing to do now is to just get the `struct device*` from a devicetree pointer, using things like `DEVICE_DT_GET`, `DEVICE_DT_GET_ANY`, or `DEVICE_DT_GET_ONE`.

Reply:
Apologies for the delayed reply,  if we remove `CONFIG_LVGL_DISPLAY_DEV_NAME` and get the device using any of the DT api , it will then go into hardware specific, also this [type](https://docs.zephyrproject.org/2.6.0/boards/shields/waveshare_epaper/doc/index.html#programming) of west build may create conflicts right?

Reply:
Thanks for the reply, but I do not understand. What conflicts are you referring to?

Reply:
I thought, if --DSHIELD overlay and device name used inside main.c will create conflicts but it wont. 
[here](https://github.com/zephyrproject-rtos/zephyr/pull/38306/commits/8208ed2d97b60ceb37474c9e2cf9db34e2c525f5) modified the requested changes